src/main/java/org/reaktivity/nukleus/oauth/internal/OAuthRealms.java

```diff
@@ -79,6 +80,20 @@ private OAuthRealms(
         this.keysByKid = keysByKid;
     }
 
+    private OAuthRealms(
+        Map<String, JsonWebKey> keysByKid,
+        boolean autoDiscoverRealms)
+    {
+        this.keysByKid = keysByKid;
+        if(autoDiscoverRealms)
+        {
+            for (Map.Entry<String, JsonWebKey> keyValue : this.keysByKid.entrySet())
+            {
+                realmsByName.computeIfAbsent(keyValue.getKey(), OAuthRealm::new);
+            }
+        }
```

Comment:
Add a method called `supplyRealm` as follows:
```
OAuthRealm supplyRealm(
    String realmName)
{
    return realmsByName.computeIfAbsent(realmName, OAuthRealm::new);
}
```
...then use it in `OAuthRealms` constructor.
```
if (autoDiscoverRealms)
{
    keysById.keySet().forEach(this::supplyRealm);
}
```